### DISCUSSION TITLE
Is there any way to integrate some extensions written by C++?

### DISCUSSION THREAD

**Discussion by FaresX** [Gnimuc/CImGui.jl#111] [Q&A]
imgui has many useful extensions but most of them are written by C++. I wonder if I can integrate them to my own applications by some certain method. I am   not familiar with C++ and the only thing I can do is to rewrite the source codes that only works for short projects. I could learn anything and wonder about the shortest path.

**Comment by JamesWrigley** [comment]
So in theory yes, but you'll have to create C bindings for the extension. The best way is probably to use [CxxWrap.jl](https://github.com/JuliaInterop/CxxWrap.jl); for which you'll need to create a little C++ library that creates Julia-compatible C functions that call the C++ functions, and a Julia module that uses those bindings (which you can use from your ImGui program).

A more hard-core alternative would be to extend [dear_bindings](https://github.com/dearimgui/dear_bindings) to work with the extensions that you want to use, and use that to create C bindings, which you can put in a JLL and call from Julia. This is similar to how the bindings for ImGui are generated, and it has the advantage that other languages could also use those bindings.

**Reply by JamesWrigley** [reply]
Oh, and this was just registered: https://github.com/grasph/WrapIt.jl
No idea how well it works but it might be worth trying out, especially since most imgui extensions have a relatively small API.
